data/test/scenarios/wml_tests/ScenarioWML/EventWML/events-test_filter_attack_type.cfg

```diff
@@ -84,8 +86,46 @@
         {ASSERT ({VARIABLE_CONDITIONAL triggers equals 0})}
         {VARIABLE_OP triggers add 1}
     [/event]
+#enddef
+
+#####
+# API(s) being tested: [event][filter_attack]type=
+##
+# Actions:
+# Give Alice an ability that adds a damage special with addition of arcnae type to all of his weapons.
+# Give Bob resistance to -100% to arcane
+# Define events that use filter_attack matching Alice's arcane type.
+# Have Alice attack Bob.
+##
+# Expected end state:
+# An event triggers when Alice attacks during side 1's turn because type=arcane detected.
+#####
+{GENERIC_UNIT_TEST event_test_filter_attack_type (
+    {FILTER_TYPE 200}
+
     [event]
         name=turn 2
         {RETURN ({VARIABLE_CONDITIONAL triggers equals 1})}
     [/event]
 )}
+
+#####
+# API(s) being tested: [event][filter_attack]type=
+##
+# Actions:
+# Give Alice an ability that adds a damage special with addition of arcane type to all of his weapons.
+# Give Bob resistance to 50% to arcane
+# Define events that use filter_attack matching Alice's arcane type.
+# Have Alice attack Bob.
+##
+# Expected end state:
+# An event not triggers when Alice attacks during side 1's turn because type=arcane can't be detected.
+#####
+{GENERIC_UNIT_TEST event_test_filter_attack_type_fail (
+    {FILTER_TYPE 50}
+
+    [event]
+        name=turn 2
+        {RETURN ({VARIABLE_CONDITIONAL triggers equals 0})}
+    [/event]
+)}
```

Comment:
If `FILTER_TYPE` is only for use in this file then there should be an `#undef` at the end.

Reply:
done